cmd/util/ledger/migrations/staged_contracts_migration.go

```diff
@@ -148,11 +153,131 @@ func (m *StagedContractsMigration) InitMigration(
 	return nil
 }
 
+// collectAndRegisterStagedContractsFromPayloads scan through the payloads and collects the contracts
+// staged through the `MigrationContractStaging` contract.
+func (m *StagedContractsMigration) collectAndRegisterStagedContractsFromPayloads(allPayloads []*ledger.Payload) error {
+
+	// If the contracts are already passed as an input to the migration
+	// then no need to scan the storage.
+	// TODO: Maybe do this anyway?
+	if len(m.contractsByLocation) > 0 {
+		return nil
+	}
```

Comment:
Maybe extract updates from the state by default, and then refactor the flag to pass "overrides" for the staged updates?

Reply:
this is kind of what happens now. If you don't provide anything, it'll extract from the state by default.